common/src/main/java/com/wynntils/mc/render/RenderUtils.java

```diff
@@ -501,6 +498,52 @@ public static void drawTooltip(List<ClientTooltipComponent> lines, PoseStack pos
         poseStack.popPose();
     }
 
+    public static void drawTooltipAt(
+            PoseStack poseStack,
+            double renderX,
+            double renderY,
+            double renderZ,
+            List<Component> componentLines,
+            Font font,
+            boolean firstLineHasPlusHeight) {
+        poseStack.pushPose();
+
+        poseStack.translate(renderX, renderY, renderZ);
+        drawTooltip(poseStack, componentLines, font, firstLineHasPlusHeight);
+
+        poseStack.popPose();
+    }
+
+    public static int getToolTipWidth(List<ClientTooltipComponent> lines, Font font) {
+        int tooltipWidth = 0;
+
+        for (ClientTooltipComponent clientTooltipComponent : lines) {
+            int lineWidth = clientTooltipComponent.getWidth(font);
+            if (lineWidth > tooltipWidth) {
+                tooltipWidth = lineWidth;
+            }
+        }
+
+        return tooltipWidth;
+    }
+
+    public static int getToolTipHeight(List<ClientTooltipComponent> lines) {
+        int tooltipHeight = lines.size() == 1 ? -2 : 0;
+
+        for (ClientTooltipComponent clientTooltipComponent : lines) {
```

Comment:
Same here, there ought to be a sum() method. (If there isn't, Java's stream library is too crippled :().